runtime-modules/content/src/lib.rs

```diff
@@ -1435,227 +1453,320 @@ decl_module! {
+            // makes no sense to delete the first post of a reddit discussion and not delete
+            // the whole discussion. So the only way to delete the first post is to delete the
+            // thread. In any case only the thread author gets the bond if he is the actor of
+            // the deletion
+
+            let first_post_init_bloat_bond = <T as balances::Trait>::Balance::zero();
+
+            Self::transfer_to_state_cleanup_treasury_account(
+                thread_init_bloat_bond,
+                new_thread_id,
+                &account_id
+            )?;
 
-            // ensure valid channel
-            Self::ensure_channel_exists(&channel_id)?;
+            let title_hash = T::compute_hash(&params.title.encode());
+         let text_hash = T::compute_hash(&params.post_text.encode());
 
-            // ensure category_id is valid
-            Self::ensure_category_exists(&category_id)?;
 
             //
             // == MUTATION SAFE ==
             //
 
-            // Create and add new thread
-            let new_thread_id = <NextThreadId<T>>::get();
 
             // Build a new thread
             let new_thread = Thread_ {
                 title_hash: title_hash,
-                category_id: category_id,
-                author_id: forum_user_id,
-                cleanup_pay_off: BalanceOf::<T>::zero(),
+                author_id: member_id,
+                bloat_bond: thread_init_bloat_bond,
                 number_of_posts: T::PostId::zero(),
-                channel_id: channel_id,
+                channel_id: params.channel_id,
             };
 
             // Store thread
-            <ThreadById<T>>::mutate(category_id, new_thread_id, |value| {
+            <ThreadById<T>>::mutate(new_thread_id, |value| {
                 *value = new_thread.clone()
             });
 
             // Add inital post to thread
             let _ = Self::add_new_post(
                 new_thread_id,
-                category_id,
                 text_hash,
-                forum_user_id,
+                member_id,
+                account_id,
+                first_post_init_bloat_bond,
+                params.post_mutable,
             );
 
             // Update next thread id
          <NextThreadId<T>>::mutate(|n| *n += One::one());
 
-         // increase category's thread counter
-         <CategoryById<T>>::mutate(category_id, |category| category.num_direct_threads += 1);
-
             // Generate event
             Self::deposit_event(
                 RawEvent::ThreadCreated(
                     new_thread_id,
-                    forum_user_id,
-                    category_id,
-                    title_hash,
-            channel_id,
+                    member_id,
+                    params,
                 )
             );
 
             Ok(())
      }
 
-    #[weight = 10_000_000]
-    fn delete_thread(
+       #[weight = 10_000_000]
+       fn delete_thread(
```

Comment:
- When moderators delete threads, they must provide rationales. It may be that using the `ContentActor` type may get in the way of the safest representation here, so you may need a new type `DeletionActor` or something, which statically holds these rationales.
- There must be a way to signal whether this is a deletion which should be understood as archiving, i.e. where UIs can still show it, or if its deletion as its a mistake or bad, and should be hidden.